### PR TITLE
add new tests from scorex into ergo

### LINKED ISSUE #152: Checks new tests in Scorex-testkit for being plugged into Ergo

### LINKED ISSUE #352: Integration test for the long-chain synchronization
Add integration test with the two nodes:
1) First node generates a long blockchain over 500 blocks.
2) Second node synchronises with the first one.

Synchronisation should complete successfully

### PR DESCRIPTION
**Pull request by daron666**
Fixes #152

### CODE REVIEW

**Comment by mike-aksarin** [comment]
This doesn't look to have something to do with #352

**Comment by daron666** [comment]
I've added

- `NodeViewSynchronizerTests`
- `MempoolFilterPerformanceTest`
- `MempoolRemovalTest`

Haven't added
- `NodeViewHolderTests` this test assumes that modifier should be plain modifier. Our ErgoFullBlock consists of 3 different modifiers.
- `WalletSecretsTest` and  `BoxStateChangesGenerationTest` cause they wanna see type P for Proposition and Box[P] type parametrised by this proposition, meanwhile out ErgoBox haven't type parameters.

File: src/test/scala/org/ergoplatform/ErgoSanity.scala [hunk 1]
```diff
@@ -15,23 +15,22 @@ import org.scalacheck.Gen
...
 import scorex.utils.Random
 
-//todo: currently this class parametrized with UtxoState, consider DigestState as well
 trait ErgoSanity[ST <: MinimalState[PM, ST]] extends HistoryTests[TX, PM, SI, HT]
   with StateApplicationTest[PM, ST]
-  //with StateApplyChangesTest[P, TX, PM, B, ST]
   //with WalletSecretsTest[P, TX, PM]
-  //with StateRollbackTest[P, TX, PM, B, ST, SI, HT, MPool]
   with MempoolTransactionsTest[TX, MPool]
   // todo: convert MempoolFilterPerformanceTest to benchmark
```
Comment (catena2w) [inline src/test/scala/org/ergoplatform/ErgoSanity.scala]: Lets fix this todo - move performance tests to a separate benchmark that won't be started by default on push.

File: src/test/scala/org/ergoplatform/ErgoSanity.scala [hunk 2]
```diff
@@ -62,12 +61,36 @@ trait ErgoSanity[ST <: MinimalState[PM, ST]] extends HistoryTests[TX, PM, SI, HT
...
+        }
+      }
+
 }
 
 object ErgoSanity {
   type TX = ErgoTransaction
   type B = ErgoBox
   type PM = ErgoPersistentModifier
+  type CTM = BlockTransactions
```
Comment (catena2w) [inline src/test/scala/org/ergoplatform/ErgoSanity.scala]: What does `CTM` means?

Comment (daron666) [reply src/test/scala/org/ergoplatform/ErgoSanity.scala]: Carrying Transactions Modifier

**Pull request merged**